src/activeobject.h

```diff
@@ -107,27 +107,34 @@ struct BoneOverride
 
 	struct ScaleProperty
 	{
-		v3f previous;
-		v3f vector{1, 1, 1};
+		v3f previous = v3f(1.0f);
+		v3f vector = v3f(1.0f);
 		bool absolute = false;
-		f32 interp_timer = 0;
+		f32 interp_timer = 0.0f;
 	} scale;
 
 	v3f getScale(v3f anim_scale) const {
 		f32 progress = dtime_passed / scale.interp_timer;
 		if (progress > 1.0f || scale.interp_timer == 0.0f)
 			progress = 1.0f;
 		return scale.vector.getInterpolated(scale.previous, progress)
-				* (scale.absolute ? v3f(1) : anim_scale);
+				* (scale.absolute ? v3f(1.0f) : anim_scale);
 	}
 
-	f32 dtime_passed = 0;
+	f32 dtime_passed = 0.0f;
+
+	bool finishedInterpolation() const
+	{
+		return dtime_passed >= std::max(std::max(
+				position.interp_timer, rotation.interp_timer), scale.interp_timer);
```

Comment:
Is `std::max` correct here? I would assume that the parameter with the least progress should tell whether the interpolation is done. It seems to me that `0` is a special value that should be treated as "not applicable", thus needs special handling.